### ISSUE TITLE
Split command fails to detect new files (maybe like #13)

### ISSUE THREAD

**Issue by liarco** [symplify/monorepo-split-github-action#20]
Hi, I think I'm facing the same issue as #13, but even with the fix applied.

Here is the output log from my split action: https://github.com/mep-agency/web-toolkit/runs/3760142721?check_suite_focus=true#step:4:63 (it's a new opensource project so everything is public)
The output shows an untracked file, but then nothing is added to the commit.

I'm splitting my monorepo to 3 repos. One is synced correctly (the skeleton) the other 2 are missing the latest commit where I just added the LICENCE/README.md files.
Unfortunately I don't know neither how to test this locally, nor how to test a custom version of this action, with changed source code, in order to give you more insights.

Any suggestion is welcome.

Thank you for your time.

**Comment by TomasVotruba** [comment]
Hi, thanks for reporting the issue. This is very hard to test, due to many layers of bash + git + symfony process layers. Many places can go wrong :)

I'm looking at repository and it's pretty huge. I recommend starting with something small, where you have files under control. E.g. 1 file split.

Personally I would narrow that huge `.gitignore` to 5-10 lines that you really need. It looks like copy-pasted from some all ignore tutorial. It may cause troubles with releasing. It was one of my "bugs" in the past, so I share it.

**Comment by liarco** [comment]
Thank you for your feedback! I've beed testing this action in the last few months and it was working fine, except for one time where I hit the same problem, but I thought that I had broken something so I started the "split" repo from scratch, re-run the failed jobs and everything synced properly, so it's a bit strange.

I totally agree with the `.gitignore` stuff, that's built with some templates from a PHPStorm extension, but it's way better to just use the rules we need.

I'm gonna try the following steps:
 - update the `.gitignore` files and other stuff to see if having both changed and new files changes the result
 - create a smaller reproducer where I try to isolate the issue (if there's any with the action)

I'm will update the issue as soon as I have some news.

**Comment by TomasVotruba** [comment]
@liarco  Thank you :+1:

**Comment by liarco** [comment]
@TomasVotruba I think I found the problem...

I created a small reproducer and forked your action to run some tests, but I couldn't replicate the issue in any way... then I realized that I was using the `@main` tag version in order to pick my latest changes from the action and so I tried to do the same on my project: it worked as intended.

Looking at the `README.md` file from your repo, you suggest using version `2.1`, but I was using `2.0` which didn't include the fix in #13.
**But there's more:** there is no `2.1` tag in this repository, so the latest stable version seems to be the `2.0` from April which is outdated, lacks some fixes and still has arguments with `-` instead of `_`.

Am I missing anything here?
